docs/book/src/reference/formatting-pipeline.md

```diff
@@ -0,0 +1,64 @@
+# The formatting pipeline
+
+## Query matching
+
+The first step is to apply the formatting queries against an input. That
+is, each Tree-sitter query will be applied to the input's syntax tree,
+taking into account the relevant capture names against all matching
+subtrees. The result of this is effectively serialised into a list of
+atomic units -- formatting directives, defined by the matching capture
+names, and leaf node content (see also [`@leaf`](capture-names/general.md#leaf))
+-- along with the necessary metadata to drive the process.
+
+## Post-processing
```

Comment:
Actually, perhaps "Processing" would fit better with the description below than "Post-processing"?

Reply:
I think calling it "Post-processing" is fine, though, and it matches the code. I guess I just feel that the previous section needs to be more "Processing"-oriented for it to make sense to call this one "Post-processing". Should we call it "Query processing"?

Reply:
Good call :+1: Moreover, while it matches the code, "post-processing" definitely has a "this is the last thing that's done" feel to it.